includes/Generator/Order.php

```diff
@@ -790,4 +819,47 @@ protected static function calculate_refund_date( $order, $previous_refund = null
 		$random_days = wp_rand( 0, $max_days );
 		return date( 'Y-m-d H:i:s', strtotime( $base_date->date( 'Y-m-d H:i:s' ) ) + ( $random_days * DAY_IN_SECONDS ) );
 	}
+
+	/**
+	 * Generate an array of sorted dates for batch order creation.
+	 * Ensures chronological order when creating multiple orders.
+	 *
+	 * @param int   $count Number of dates to generate.
+	 * @param array $args  Arguments containing date-start and optional date-end.
+	 * @return array Sorted array of date strings (Y-m-d).
+	 */
+	protected static function generate_batch_dates( $count, $args ) {
+		$current = date( 'Y-m-d', time() );
+
+		if ( ! empty( $args['date-start'] ) && empty( $args['date-end'] ) ) {
+			$start = $args['date-start'];
+			$end   = $current;
+		} elseif ( ! empty( $args['date-start'] ) && ! empty( $args['date-end'] ) ) {
+			$start = $args['date-start'];
+			$end   = $args['date-end'];
+		} else {
+			// No date range specified, return array of current dates
+			return array_fill( 0, $count, $current );
+		}
+
+		$start_timestamp = strtotime( $start );
```

Comment:
Should we validate that `date-start` and `date-end` are in the expected format and that the start date is earlier than the end date before we use them?  Or it's already done in some other places?